clearblade/cloud/iot_v1/device_types.py

```diff
@@ -0,0 +1,430 @@
+                      last_config_ack_time= json['lastConfigAckTime'], last_config_send_time= json['lastConfigSendTime'],
+                      blocked= json['blocked'], last_error_time= json['lastErrorTime'],
+                      last_error_status_code= json['lastErrorStatus'], config= json['config'],
+                      state= json['state'], log_level= json['logLevel'], meta_data= json['metadata'],
+                      gateway_config= json['gatewayConfig'])
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def num_id(self):
+        return self._num_id
+
+    @property
+    def credentials(self):
+        return self._credentials
+
+    @property
+    def last_error_status(self):
+        return self._last_error_status_code
+
+    @property
+    def config(self):
+        return self._config
+
+    @property
+    def state(self):
+        return self._state
+
+    @property
+    def log_level(self):
+        return self._log_level
+
+    @property
+    def meta_data(self):
+        return self._meta_data
+
+    @property
+    def gateway_config(self):
+        return self._gateway_config
+
+    @property
+    def log_level(self):
+        return self._log_level
+
+#classes to mock googles request & response
+
+class DeviceState():
+    def __init__(self, updated_time: str =None, binary_data:str = None) -> None:
+        self._updated_time = updated_time
+        self._binary_data = binary_data
+
+    @property
+    def updated_time(self):
+        return self._updated_time
+
+    @property
+    def binary_data(self):
+        return self._binary_data
+
+    @staticmethod
+    def from_json(response_json):
+        return DeviceState(updated_time=get_value(response_json, 'updateTime'),
+                           binary_data=get_value(response_json, 'binaryData'))
+
+class Request():
+    def __init__(self, name) -> None:
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+class SendCommandToDeviceRequest(Request):
+    def __init__(self, name: str = None,
+                binary_data: bytes = None,
+                subfolder: str = None) -> None:
+        super().__init__(name)
+        self._binary_data = binary_data
+        self._subfolder = subfolder
+
+    @property
+    def binary_data(self):
+        return self._binary_data
+
+    @property
+    def sub_folder(self):
+        return self._subfolder
+
+class CreateDeviceRequest(Request):
+    def __init__(self, name: str = None,
+                       device: Device = None) -> None:
+        super().__init__(name)
+        self._device = device
+
+    @property
+    def device(self):
+        return self._device
+
+class ModifyCloudToDeviceConfigRequest(Request):
+    def __init__(self, name:str = None ,
+                 version_to_update: int = -1,
```

Comment:
modifyCloudToDeviceConfig failing due to incorrect data type. version_to_update should be a string.

Reply:
let the default value for this be an empty string